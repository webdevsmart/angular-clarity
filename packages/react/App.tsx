import React from 'react';
import { CdsAlert, CdsAlertActions, CdsAlertGroup } from './src/alert';
import { CdsButton, CdsButtonType } from './src/button';
import { CdsBadge } from './src/badge';
import { CdsIcon } from './src/icon';
import { CdsModal, CdsModalActions, CdsModalContent, CdsModalHeader } from './src/modal';
import { CdsTag } from './src/tag';
import { ClarityIcons, userIcon, timesIcon } from '@clr/core/icon';

ClarityIcons.addIcons(userIcon, timesIcon);

interface AppState {
  modalOpen: boolean;
}

export default class App extends React.Component<{}, AppState> {
  buttonRef: any;

  constructor(props: any) {
    super(props);
    this.state = { modalOpen: false };
    this.buttonRef = React.createRef<CdsButtonType>();
  }

  componentDidMount() {
    this.buttonRef.current.nativeElement.then((element: any) => {
      element.focus();
    });
  }

  handleOnCloseChange(e: any) {
    console.log(e);
  }

  render() {
    const isModalOpen = this.state.modalOpen;
    return (
      <div>
        <h1>Rendered by React!</h1>
        <CdsButton ref={this.buttonRef} onClick={() => this.setState({ modalOpen: true })}>
          Open Modal
        </CdsButton>
        {isModalOpen ? (
          <CdsModal onCloseChange={() => this.setState({ modalOpen: false })}>
            <CdsModalHeader>
              <h3 cds-text="title">My Modal</h3>
            </CdsModalHeader>
            <CdsModalContent>
              <div cds-layout="vertical gap:md p-y:xs">
                <p cds-text="body">Lorem Ipsum</p>
              </div>
            </CdsModalContent>
            <CdsModalActions>
              <div cds-layout="horizontal gap:sm align:right">
                <CdsButton onClick={() => this.setState({ modalOpen: false })} action="outline">
                  Cancel
                </CdsButton>
                <CdsButton onClick={() => this.setState({ modalOpen: false })}>Ok</CdsButton>
              </div>
            </CdsModalActions>
          </CdsModal>
        ) : (
          <br />
        )}
        <h2>Light Alerts</h2>
        <CdsAlert status="info" onCloseChange={e => this.handleOnCloseChange(e)}>
          Foobar
        </CdsAlert>
        <CdsAlert status="success">Foo</CdsAlert>
        <CdsAlert status="warning">Bar</CdsAlert>
        <CdsAlert status="danger">Baz</CdsAlert>

        <h2>Alert Group</h2>
        <CdsAlertGroup status="info">
          <CdsAlert closable>This example is a closable alert inside an alert group with a status of "info".</CdsAlert>
          <CdsAlert closable>
            Foobar
            <CdsAlertActions>
              <CdsButton>Fix</CdsButton>
            </CdsAlertActions>
          </CdsAlert>
        </CdsAlertGroup>
        <CdsAlertGroup status="success">
          <CdsAlert closable>
            This example is a closable alert inside an alert group with a status of "success".
          </CdsAlert>
          <CdsAlert closable>
            Foobar
            <CdsAlertActions>
              <CdsButton>Fix</CdsButton>
            </CdsAlertActions>
          </CdsAlert>
        </CdsAlertGroup>
        <CdsAlertGroup status="warning">
          <CdsAlert closable>
            This example is a closable alert inside an alert group with a status of "warning".
          </CdsAlert>
          <CdsAlert closable>
            Foobar
            <CdsAlertActions>
              <CdsButton>Fix</CdsButton>
            </CdsAlertActions>
          </CdsAlert>
        </CdsAlertGroup>
        <CdsAlertGroup status="danger">
          <CdsAlert closable>
            This example is a closable alert inside an alert group with a status of "danger".
          </CdsAlert>
          <CdsAlert closable>
            Foobar
            <CdsAlertActions>
              <CdsButton>Fix</CdsButton>
            </CdsAlertActions>
          </CdsAlert>
        </CdsAlertGroup>

        <h2>Banner Alerts</h2>
        <CdsAlertGroup type="banner">
          <CdsAlert status="info">
            Foobar
            <CdsAlertActions>
              <CdsButton>Fix</CdsButton>
            </CdsAlertActions>
          </CdsAlert>
        </CdsAlertGroup>
        <CdsAlertGroup type="banner">
          <CdsAlert status="warning">
            Bar
            <CdsAlertActions>
              <CdsButton>Fix</CdsButton>
            </CdsAlertActions>
          </CdsAlert>
        </CdsAlertGroup>
        <CdsAlertGroup type="banner">
          <CdsAlert status="danger">
            Baz
            <CdsAlertActions type="banner">
              <CdsButton>Fix</CdsButton>
            </CdsAlertActions>
          </CdsAlert>
        </CdsAlertGroup>

        <h2>Buttons</h2>
        <CdsButton status="primary">primary</CdsButton>
        <CdsButton status="success">success</CdsButton>
        <CdsButton status="danger">danger</CdsButton>
        <CdsButton status="danger" disabled>
          disabled
        </CdsButton>

        <h2>Badge</h2>
        <CdsBadge status="info">2</CdsBadge>
        <CdsBadge status="success">3</CdsBadge>
        <CdsBadge status="warning">12</CdsBadge>
        <CdsBadge status="danger">15</CdsBadge>
        <CdsBadge color="gray">1</CdsBadge>
        <CdsBadge color="purple">1</CdsBadge>
        <CdsBadge color="blue">15</CdsBadge>
        <CdsBadge color="orange">2</CdsBadge>
        <CdsBadge color="light-blue">3</CdsBadge>

        <h2>Tags</h2>
        <CdsTag readonly status="info">
          Info
        </CdsTag>
        <CdsTag readonly status="success">
          Success
        </CdsTag>
        <CdsTag readonly status="warning">
          Warning
        </CdsTag>
        <CdsTag readonly status="danger">
          Danger
        </CdsTag>
        <br />
        <br />
        <CdsTag readonly color="gray">
          Austin <CdsBadge>1</CdsBadge>
        </CdsTag>
        <CdsTag readonly color="purple">
          New York <CdsBadge>2</CdsBadge>
        </CdsTag>
        <CdsTag readonly color="blue">
          Palo Alto <CdsBadge>3</CdsBadge>{' '}
        </CdsTag>
        <CdsTag readonly color="orange">
          San Francisco <CdsBadge>12</CdsBadge>
        </CdsTag>
        <CdsTag readonly color="light-blue">
          Seattle <CdsBadge>15</CdsBadge>
        </CdsTag>

        <h2>Icons</h2>
        <CdsIcon size="lg" shape="user"></CdsIcon>
        <CdsIcon size="lg" shape="user" badge="info"></CdsIcon>
        <CdsIcon size="lg" shape="user" badge="success"></CdsIcon>
        <CdsIcon size="lg" shape="user" badge="danger"></CdsIcon>
        <CdsIcon size="lg" shape="user" badge="warning-triangle"></CdsIcon>
        <br />
        <br />
        <CdsIcon size="lg" shape="user" solid></CdsIcon>
        <CdsIcon size="lg" shape="user" solid badge="info"></CdsIcon>
        <CdsIcon size="lg" shape="user" solid badge="success"></CdsIcon>
        <CdsIcon size="lg" shape="user" solid badge="danger"></CdsIcon>
        <CdsIcon size="lg" shape="user" solid badge="warning-triangle"></CdsIcon>
      </div>
    );
  }
}
