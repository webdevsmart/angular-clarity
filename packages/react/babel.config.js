// used by our jest test suite; not used for running the dev app
module.exports = {
  presets: [['@babel/preset-env', { targets: { esmodules: true } }]],
};
