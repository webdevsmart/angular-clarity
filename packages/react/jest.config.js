module.exports = {
  roots: ['./'],
  globals: {
    'ts-jest': {
      tsConfig: 'tsconfig.test.json',
    },
  },
  modulePathIgnorePatterns: ['<rootDir>/dist/'],
  moduleNameMapper: {
    '@clr/(.*)': '<rootDir>/../../dist/$1',
  },
  setupFilesAfterEnv: ['./jest.setup.ts'],
  snapshotSerializers: ['enzyme-to-json/serializer'],
  testMatch: ['**/__tests__/**/*.+(ts|tsx|js)', '**/?(*.)+(test).+(ts|tsx|js)'],
  transform: {
    '^.+\\.(js|jsx)$': 'babel-jest',
    '^.+\\.(ts|tsx)$': 'ts-jest',
  },
  transformIgnorePatterns: ['node_modules/(?!@clr/core|lit-element|lit-html|ramda)'],
};
