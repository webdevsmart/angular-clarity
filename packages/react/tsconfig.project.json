{
  "extends": "./tsconfig.lib.json",
  "include": ["./src"],
  "references": [
    { "path": "./src/converter/entrypoint.tsconfig.json" },
    { "path": "./src/alert/entrypoint.tsconfig.json" },
    { "path": "./src/badge/entrypoint.tsconfig.json" },
    { "path": "./src/button/entrypoint.tsconfig.json" },
    { "path": "./src/icon/entrypoint.tsconfig.json" },
    { "path": "./src/modal/entrypoint.tsconfig.json" },
    { "path": "./src/tag/entrypoint.tsconfig.json" }
  ]
}
