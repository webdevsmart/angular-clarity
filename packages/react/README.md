# Clarity React Wrappers for Core Web Components

Clarity Web Components is a suite of Web Components from the [Clarity Design System](https://clarity.design).

## Installing Clarity Web Components

1.  First, install the Clarity Core package from npm.

    ```bash
    npm install @clr/react --save
    ```

2.  Import desired Web Component into your JavaScript or TypeScript

    ```typescript
    import { CdsModal, CdsModalActions, CdsModalContent, CdsModalHeader } from '@clr/react/modal';
    ```

3.  Use Web Component in react

    ```jsx
    {
      /*
      Example of a modal web component in React with React Shim
      - size - attribute style hook
      - open - setting the 'open' property on the element
      - openChange - listen for the `openChange` custom event
    */
    }
    <CdsModal onCloseChange={this.handleCloseModalClick.bind(this, false)}>
      <CdsModalHeader>
        <h3 cds-text="title">My Modal</h3>
      </CdsModalHeader>
      <CdsModalContent>
        <div cds-layout="vertical gap:md p-y:xs">
          <p cds-text="body">Lorem Ipsum</p>
        </div>
      </CdsModalContent>
      <CdsModalActions>
        <div cds-layout="horizontal gap:sm align:right">
          <CdsButton onClick={this.handleCloseModalClick.bind(this, false)} action="outline">
            Cancel
          </CdsButton>
          <CdsButton onClick={this.handleCloseModalClick.bind(this, false)}>Ok</CdsButton>
        </div>
      </CdsModalActions>
    </CdsModal>;
    ```
