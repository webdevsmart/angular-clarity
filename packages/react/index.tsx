import React from 'react';
import ReactDOM from 'react-dom';
import 'normalize.css/normalize.css';
import '@clr/core/global.min.css';
import '@clr/city/css/bundles/default.min.css';
import App from './App';

ReactDOM.render(<App />, document.getElementById('root'));
