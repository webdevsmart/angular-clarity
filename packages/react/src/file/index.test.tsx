import * as React from 'react';
import { mount, shallow } from 'enzyme';
import { CdsFile } from './index';

describe('CdsFile', () => {
  it('renders', () => {
    const wrapper = shallow(
      <div>
        <CdsFile layout="vertical">
          <label>label</label>
          <input type="file" multiple />
        </CdsFile>
      </div>
    );
    const renderedComponent = wrapper.find('ReactWrapperComponent');
    expect(renderedComponent.html()).toMatch(/label/);
  });

  it('snapshot', () => {
    const wrapper = mount(
      <div>
        <CdsFile layout="vertical">
          <label>label</label>
          <input type="file" multiple />
        </CdsFile>
      </div>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
