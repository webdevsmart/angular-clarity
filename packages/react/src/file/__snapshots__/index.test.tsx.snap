// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CdsFile snapshot 1`] = `
<div>
  <ReactWrapperComponent
    layout="vertical"
  >
    <cds-file>
      <label>
        label
      </label>
      <input
        multiple={true}
        type="file"
      />
    </cds-file>
  </ReactWrapperComponent>
</div>
`;
